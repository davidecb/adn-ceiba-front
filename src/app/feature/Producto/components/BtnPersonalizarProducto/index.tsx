import * as PropTypes from 'prop-types';
import * as React from 'react';
import { Button } from 'app/shared/components/Button';
import { Producto } from '../../models/Producto';

interface BtnPersonalizarProductoProps {
  onPersonalizar: (producto: Producto) => any;
  producto: Producto;
}

export const BtnPersonalizarProducto: React.FC<BtnPersonalizarProductoProps> = ({
  onPersonalizar,
  producto,
}) => {
  const handlePersonalizar = () => onPersonalizar(producto);
  return (
    <Button onClick={handlePersonalizar}>
      <span>
        personalizar
      </span>
    </Button>
  );
};

BtnPersonalizarProducto.propTypes = {
  producto: PropTypes.shape({
    id: PropTypes.number.isRequired,
    nombre: PropTypes.string.isRequired,
    costo: PropTypes.number.isRequired,
    tiempo: PropTypes.number.isRequired,
    imagen: PropTypes.string.isRequired
  }).isRequired,
  onPersonalizar: PropTypes.func.isRequired,
};
