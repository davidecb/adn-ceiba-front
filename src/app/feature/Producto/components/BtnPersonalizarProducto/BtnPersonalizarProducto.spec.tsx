import * as React from 'react';
import { RenderResult, render } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { BtnPersonalizarProducto } from './../BtnPersonalizarProducto';

describe('BtnPersonalizarProducto Test', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof BtnPersonalizarProducto> & {
    onPersonalizar: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      producto: {
        id: 1,
        nombre: 'producto testing',
        costo: 10000,
        tiempo: 30,
        imagen: 'imagen.jpg'
      },
      onPersonalizar: stub(),
    };
    componentWrapper = render(<BtnPersonalizarProducto {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });
});

export {};
