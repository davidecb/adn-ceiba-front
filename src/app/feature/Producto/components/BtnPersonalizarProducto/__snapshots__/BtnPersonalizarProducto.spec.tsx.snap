// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnPersonalizarProducto Test should match snapshot 1`] = `
<div>
  <button
    class="sc-AxjAm dmxFrE"
  >
    <span>
      personalizar
    </span>
  </button>
</div>
`;
